qemu/tests/ple_test.py

```diff
@@ -0,0 +1,71 @@
+import re
+import logging
+
+from avocado.utils import cpu
+from virttest import env_process
+from virttest import error_context
+from virttest import arch
+from virttest.utils_kernel_module import KernelModuleHandler
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    ple test:
+    1) Check if ple is enabled on host, if not, enable it
+    2) Boot guest and run unixbench inside guest
+    3) Record benchmark scores and shutdown guest
+    4) Disable ple on host
+    5) Boot guest and run unixbench inside guest again
+    6) Compare benchmark scores with step 3)
+    7) Restore env, enable ple
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def run_unixbench(cmd):
+        """
+        Run unixbench inside guest, return benchmark scores
+        """
+        error_context.context("Run unixbench inside guest", logging.info)
+        output = session.cmd_output_safe(cmd, timeout=4800)
+        scores = re.findall(r"System Benchmarks Index Score\s+(\d+\.?\d+)",
+                            output)
+        return [float(i) for i in scores]
+
+    error_context.context("Enable ple on host", logging.info)
```

Comment:
Wow, you change the rule?  No matter what the initinal value is for the host, just set the expected value by force ?

Reply:
Yeah, as long as it is not 0, ple is enabled. So it's ok to set a specified ple_value to test when it's enabled.